src/main/xsl/preprocess/topicpullImpl.xsl

```diff
@@ -100,31 +187,36 @@ mode="topicpull:figure-linktext" and mode="topicpull:table-linktext"
       <!--@type|@format|@scope|@importance|@role-->
 
       <!--need to create type, format, scope variables regardless of whether they exist, for passing as a parameter to getstuff template-->
-      <xsl:variable name="type" as="xs:string">
-        <xsl:call-template name="topicpull:inherit"><xsl:with-param name="attrib">type</xsl:with-param></xsl:call-template>
-      </xsl:variable>
-      <xsl:variable name="format" as="xs:string">
-        <xsl:call-template name="topicpull:inherit"><xsl:with-param name="attrib">format</xsl:with-param></xsl:call-template>
-      </xsl:variable>
-      <xsl:variable name="scope" as="xs:string">
-        <xsl:call-template name="topicpull:inherit"><xsl:with-param name="attrib">scope</xsl:with-param></xsl:call-template>
-      </xsl:variable>
-
-      <xsl:if test="not(@type) and $type!='#none#'">
+      <xsl:variable name="type" as="xs:string?" select="dita-ot:get-link-target-type(.)"/>
+      
+      <xsl:variable name="format" as="xs:string?" select="dita-ot:get-link-format(.)"/>
+      <xsl:variable name="scope" as="xs:string?" select="dita-ot:get-link-scope(.)"/>
+      
+      <xsl:if test="not(@type) and $type">
         <xsl:attribute name="type"><xsl:value-of select="$type"/></xsl:attribute>
       </xsl:if>
-      <xsl:if test="not(@format) and $format!='#none#'">
+      <xsl:if test="not(@format) and $format">
         <xsl:attribute name="format"><xsl:value-of select="$format"/></xsl:attribute>
       </xsl:if>
-      <xsl:if test="not(@scope) and $scope!='#none#'">
+      <xsl:if test="not(@scope) and $scope">
         <xsl:attribute name="scope"><xsl:value-of select="$scope"/></xsl:attribute>
       </xsl:if>
 
       <xsl:if test="not(@importance)">
-        <xsl:apply-templates select="." mode="topicpull:inherit-and-set-attribute"><xsl:with-param name="attrib">importance</xsl:with-param></xsl:apply-templates>
+        <xsl:variable name="importance" as="xs:string?"
+          select="dita-ot:get-inherited-attribute-value(., 'importance', ())"
+        />
+        <xsl:if test="$importance">
+          <xsl:attribute name="importance" select="$importance"/>
+        </xsl:if>
       </xsl:if>
       <xsl:if test="not(@role)">
-        <xsl:apply-templates select="." mode="topicpull:inherit-and-set-attribute"><xsl:with-param name="attrib">role</xsl:with-param></xsl:apply-templates>
+        <xsl:variable name="role" as="xs:string?"
+          select="dita-ot:get-inherited-attribute-value(., 'role', ())"
+        />
```

Comment:
The formatting style of pushing `/>` to a new line is something that's not used in almost at all on this code base. Thus please stick to the common convention of `/>` on the same line as the last attribute or element name.